packages/chain/contracts/Eth_broker.sol

```diff
@@ -417,14 +418,17 @@ contract Eth_broker {
             "DAI required for trade above max"
         );
         // Swapping sent ETH for exact amount of DAI needed
-        router_.swapETHForExactTokens.value(msg.value)(
+        uint256[] memory swapOutputs = new uint[](2);
+     
+        swapOutputs = router_.swapETHForExactTokens.value(msg.value)(
             daiNeeded, 
             getPath(true), 
             address(this), 
             _deadline
         );
         // Getting the amount of ETH received
         ethReceived = address(this).balance;
+        ethSpent = swapOutputs[0];
```

Comment:
You never use this variable. Why burdening yourself with assigning it (also to initialize and assign the array on lines 421 and 423)

Reply:
To match the uniswap broker, the issue with testing was that the values for swapOutputs wasn't being returned in the mocks